source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -46,11 +46,9 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
       ssl_curves_(stat_name_set_->add("ssl.curves")),
       ssl_sigalgs_(stat_name_set_->add("ssl.sigalgs")) {
   const auto tls_certificates = config.tlsCertificates();
-// Use a single context for certificates instead of one context per certificate as
-// openssl handles the multi cert case automatically in the handshake
-  tls_contexts_.resize(1);
 
-  for (auto& ctx : tls_contexts_) {
+  auto& ctx = tls_context_;
+  {
```

Comment:
Here and below: could you either remove the braces left after ```for``` loops, or add TODOs to deal with them later?

Reply:
Originally left the {} in place to minimise risk of failure, on further analysis removed and re-tested.